rids/rules/tls_matcher.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2022 Jigsaw Operations LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from rids.event import Event
+from rids.monitors import tls_monitor
+
+
+class TlsMatcher:
+  """Index over TLS-related rules to match against."""
+
+  def __init__(self):
+    self.allowed_sni_port = set()
+
+  def add_allowed_sni(self, allowed_sni: str, expected_port: int) -> None:
```

Comment:
Since this is more than the SNI, I'd name it `add_allowed_endpoint`. It can be in a follow up PR.

Reply:
sounds good.. I had briefly named it endpoint but since that term can mean multiple things (depending on network layer, or even sometimes applications have different meanings for the term), I renamed it.  I had been using sni_port in many places, I'm kind of surprised the _port got dropped here.  If you think endpoint makes sense here, that's good enough for me.